### PR TITLE
[1346] add mcb config system

### PR DESCRIPTION
**Pull request by misaka**
https://trello.com/c/11v4CXhF/1346-mcb-provider-optin-should-audit-which-user-ran-the-command

### Context

When admins make changes via `mcb` we want to ensure there is an audit trail for who did the changes. This also requires some kind of a config system to keep track of the admin's email address, and which will also be useful for future features.

### Changes proposed in this pull request

This PR introduces a config mechanism that will be used in a near-future PR to add the audit stuff. It also does a little bit of specs tidying up that fell out of the work of adding the config stuff.

### Guidance to review

Config mechanism has it's own PR since it's grown a bit big. Another PR for auditing system to follow later.

### CODE REVIEW

**Comment by misaka** [comment]
['mcb provider optin' should audit which user ran the command](https://trello.com/c/11v4CXhF/1346-mcb-provider-optin-should-audit-which-user-ran-the-command)

**Comment by misaka** [comment]
Whoops. Included too much in this PR. Re-cutting.

File: bin/mcb [hunk 1]
```diff
@@ -36,6 +37,11 @@ $mcb = Cri::Command.define do
     puts cmd.help
     exit 0
   end
+
+  option :c, :config, 'use provided config file',
```
Comment (defong) [inline bin/mcb]: can you throw in an example config file

Comment (misaka) [reply bin/mcb]: Not really meant to be used this way, as in ... we shouldn't have to add the config stuff by hand. mcb commands will use this to save stuff, and if the user has to do something special they can prompt the user to do so (e.x. the new audit command will [ask the user to add their email](https://github.com/DFE-Digital/manage-courses-backend/pull/324/files#diff-fabca55fa119b555a7942669c431eafeR212))

**Pull request merged**